doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units, storage units, and network transfer units. A strawman
+proposal: compute units are virtual machine-hours, storage units are
+gigabyte-months, and network transfer units are gigabytes. Only outbound
+network traffic is billed.
+
+We expect metering the compute usage of clusters, the storage usage of sources
+and tables, and outbound network traffic generated by sinks and query responses,
+to be straightforward. What is not clear is how to bill for the overhead of
+other system components.
+
+### Region
+
+When an account is created, it does not have any regions. The user's first step
+after registering for a new account is to choose one region to activate.
+
+Additional regions can be enabled via a "+ Enable new region" button in the
+region selector or somesuch in the web UI, or via the [global API](#global-api).
+
+To prevent disaster, regions cannot be deactivated. Users who wish to deactivate
+a region can contact support.
+
+The initially supported regions will be the same regions as supported today
+by Materialize Cloud:
+
+  * aws:us-east-1
+  * aws:eu-west-1
+
+### Cluster
+
+Clusters are created within a region via a new `CREATE CLUSTER` SQL statement:
+
+```sql
+CREATE CLUSTER <name> SIZE {xs|s|m|l|xl} [, REPLICATION FACTOR <n>];
+```
+
+Creating a cluster allocates persistent compute resources (e.g., EC2 instances)
+and begins billing the account for those resources. The SQL statement returns
+successfully as soon as the request is received.
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. Any requests to the cluster will be
+transparently routed to any live replica, so the cluster can tolerate failures
+of `n - 1` replicas before it becomes unavailable.
+
+A cluster may be horizontally scaled across multiple machines. The scaling is
+deliberately hidden from users behind the opaque notion of "size".
+
+The list of clusters and the status of each can be observed via the new `SHOW
+CLUSTERS` statement or the new `mz_clusters` system catalog table. The exact
+output of these commands is TODO, but they will include the provisioning status
+of the EC2 instances.
+
+It is possible to create indexes in a cluster before it is provisioned, but
+attempts to `SELECT` from these indexes will fail until the cluster is
+provisioned.
+
+Clusters are dropped via a new `DROP CLUSTER` SQL statement:
+
+```sql
+DROP CLUSTER <name> [CASCADE];
+```
+
+Clusters cannot be dropped while they contain any indexes unless `CASCADE`
+is specified.
+
+The active cluster for a session is configured via a new `cluster` session
+variable:
+
+```sql
+SET cluster = <name>;
+```
+
+Users can configure a default cluster for their sessions. How this default is
+configured is TODO.
+
+Access to a cluster is gated by SQL permission grants. Users cannot execute
```

Comment:
CREATE privileges are probably also required (since this causes the $ spend to increase) to create clusters